enable distributed tests on sandcastle

zhaojuanmao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #67858

enable distributed tests on sandcastle, look like other tests like c10d_nccl, shardedTensor and etc can use spawn start method by calling _spawning_process()

Differential Revision: [D32115780](https://our.internmc.facebook.com/intern/diff/D32115780/)

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

zhaojuanmao committed:
enable distributed tests on sandcastle

enable distributed tests on sandcastle, look like other tests like c10d_nccl, shardedTensor and etc can use spawn start method by calling _spawning_process()

Differential Revision: [D32115780](https://our.internmc.facebook.com/intern/diff/D32115780/)

[ghstack-poisoned]

mrshenli reviewed (approved):
Curious, is there any recent change in sandcastle that stabilized spawn support?

zhaojuanmao commented:
> Curious, is there any recent change in sandcastle that stabilized spawn support?

I did not know why it was disabled before, all other tests like c10d, sharded tensors, FSDP and etc are using spawn processes, they are fine. So trying to enable them back in sandcastle

zhaojuanmao committed:
Update on "enable distributed tests on sandcastle"


enable distributed tests on sandcastle, look like other tests like c10d_nccl, shardedTensor and etc can use spawn start method by calling _spawning_process()

Differential Revision: [D32115780](https://our.internmc.facebook.com/intern/diff/D32115780/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]
